c7n/resources/cloudfront.py

```diff
@@ -65,3 +67,147 @@ def get_dimensions(self, resource):
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: streaming-distribution-delete
+                resource: streaming-distribution
+                filters:
+                  - type: value
+                    key: S3Origin.OriginAccessIdentity
+                    value: ''
+                actions:
+                  - type: disable
+    """
+    schema = type_schema('disable')
+
+    permissions = ("streaming-distribution:GetStreamingDistributionConfig",
+                   "streaming-distribution:UpdateStreamingDistribution",)
+
+    def process(self, distributions):
+        with self.executor_factory(max_workers=2) as w:
+            list(w.map(self.process_distribution, distributions))
+
+    def process_distribution(self, distribution):
+        client = local_session(
+            self.manager.session_factory).client(self.manager.get_model().service)
+        try:
+            res = client.get_streaming_distribution_config(
+                Id=distribution[self.manager.get_model().id])
+            res['StreamingDistributionConfig']['Enabled'] = False
+            res = client.update_streaming_distribution(
+                Id=distribution[self.manager.get_model().id],
+                IfMatch=res['ETag'],
+                StreamingDistributionConfig=res['StreamingDistributionConfig']
+            )
+        except Exception as e:
+            self.log.warning(
+                "Exception trying to disable Distribution: %s error: %s",
+                distribution['ARN'], e)
+            return
+
+
+@Distribution.action_registry.register('set-ssl')
+class DistributionSSLAction(BaseAction):
+    """Action to set mandatory https-only on a Distribution
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: distribution-set-ssl
+                resource: distribution
+                filters:
+                  - type: value
+                    key: CacheBehaviors.Items[].ViewerProtocolPolicy
+                    value: allow-all
+                    op: contains
+                actions:
+                  - type: set-ssl
+    """
+    schema = type_schema('set-ssl')
+    permissions = ("distribution:GetDistributionConfig",
+                   "distribution:UpdateDistribution",)
+
+    def process(self, distributions):
+        with self.executor_factory(max_workers=2) as w:
+            list(w.map(self.process_distribution, distributions))
+
+    def process_distribution(self, distribution):
+        client = local_session(
+            self.manager.session_factory).client(self.manager.get_model().service)
+        try:
+            res = client.get_distribution_config(
+                Id=distribution[self.manager.get_model().id])
+            etag = res['ETag']
+            dc = res['DistributionConfig']
+
+            for item in dc['CacheBehaviors'].get('Items', []):
+                item['ViewerProtocolPolicy'] = 'https-only'
```

Comment:
make this configurable via policy param, with https-only  as default, also this should ideally allow for configuration on setting OriginSslProtocols (ie tls 1.2 etc).